lib/code_corps/guardian.ex

```diff
@@ -0,0 +1,40 @@
+defmodule CodeCorps.Guardian do
+  use Guardian, otp_app: :code_corps
+
+  alias CodeCorps.{Project, Repo, User}
+
+  def subject_for_token(project = %Project{}, _claims) do
+    {:ok, "Project:#{project.id}"}
+  end
+  def subject_for_token(user = %User{}, _claims) do
+    {:ok, "User:#{user.id}"}
+  end
+  def subject_for_token(_, _) do
+    {:error, :unknown_resource_type}
+  end
+
+  def resource_from_claims(%{"sub" => sub}), do: resource_from_sub(sub)
+  def resource_from_claims(_), do: {:error, :unknown_resource_type}
```

Comment:
Not sure if this can happen.

Reply:
I believe this _can_ happen, but the `:error` reason is probably not accurate. Maybe `:missing_resource`?